NugetLightRequest.cs

```diff
@@ -1909,5 +1930,169 @@ private IEnumerable<PackageItem> SearchForPackages(PackageSource source, string
+                    process.Start();
+                    StreamReader reader = process.StandardOutput;
+
+                    while (!reader.EndOfStream)
+                    {
+                        string line = reader.ReadLine();
+                        if (Regex.IsMatch(line, @"installationPath"))
+                        {
+                            // Match all chars after 'installationPath:'
+                            Match vsInstallPathMatch = Regex.Match(line, @"(?<=\: ).*");
+                            vsInstallationPath = vsInstallPathMatch.ToString();
+                            break;
+                        }
+                    }
+                }
+                catch (Exception e)
+                {
+                    Console.WriteLine(e.Message);
+                }
+                finally
+                {
+                    process.Close();
+                }
+                // Then use the installation path discovered by vswhere.exe to create the path to search for credential provider
+                // ex: "C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise" + "\Common7\IDE\CommonExtensions\Microsoft\NuGet\Plugins\CredentialProvider.Microsoft\CredentialProvider.Microsoft.exe"
+                if (vsInstallationPath.IsNullOrEmpty())
+                {
+                    throw new ArgumentNullException("vsInstallationPath");
+                }
+                credProviderPath = vsInstallationPath + "\\Common7\\IDE\\CommonExtensions\\Microsoft\\NuGet\\Plugins\\CredentialProvider.Microsoft\\CredentialProvider.Microsoft.exe";
+            }
+
+            // Using a process to run CredentialProvider.Microsoft.exe with arguments -V verbose -U query (and -IsRetry when appropriate)
+            // See: https://github.com/Microsoft/artifacts-credprovider
+            Process proc = new Process();
+            proc.StartInfo.FileName = credProviderPath;
+            var arguments = "-V verbose -U " + query;
+            if (isRetry)
+            {
+                arguments = "-I " + arguments;
+            }
+            proc.StartInfo.Arguments = arguments;
+            // Need to redirect to save tokens
+            proc.StartInfo.RedirectStandardOutput = true;
+            proc.StartInfo.CreateNoWindow = false;
+            proc.StartInfo.UseShellExecute = false;
+
+            try
+            {
+                proc.Start();
+                StreamReader reader = proc.StandardOutput;
+
+                // Write the redirected output to console
+                while (!reader.EndOfStream)
+                {
+                    string line = reader.ReadLine();
+                    // Need to decide the level of verbosity that should be displayed (or whether a user specified flag should determine this)
+                    if (Regex.IsMatch(line, @"Minimal"))
+                    {
+                        // Minimal will always print to console
+                        Console.WriteLine(line);
```

Comment:
we should avoid Console.WriteLine and put things via request object instead

Reply:
This is actually printing the device flow to the console